### PR TITLE
Multi Record Fix + Absence Table Count and ID Obfuscation

### LINKED ISSUE #90: Record over riding onto another record
Submitter: Nic Smith
Email: nismith@lincoln.ac.uk
  I have created a record for George Onoufriou as he has withdrawn from MSc.  When you go into his record (MSc) it looks like the MPhil/PhD record has overwritten it.

### PR DESCRIPTION
**Pull request by pet1330**
Closes #90 & #91

**Please note:** this only fixes issues with student records going forward, any records which have already been incorrectly modified will need to be manually updated.

### CODE REVIEW

**Comment by pet1330** [comment]
It's only the tests that are failing, I'll fix them tomorrow. @marc-hanheide can you play with this version on the staging server and check it works as expected?

File: app/Http/Controllers/StudentRecordController.php [hunk 1]
```diff
@@ -128,7 +128,6 @@ public function show(Request $request, Student $student, StudentRecord $record)
             return $this->absences_controls($this->absences($record), $student)->make(true);
         }
 
-        $record = $student->record();
```
Comment (marc-hanheide) [inline app/Http/Controllers/StudentRecordController.php]: Was confused by this at first, but get it now. So, just some undeleted legacy code?

File: app/Models/Student.php [hunk 2]
```diff
@@ -22,76 +22,6 @@ public function records()
         return $this->hasMany(StudentRecord::class);
     }
 
-    public function record()
-    {
-        return $this->records->last();
```
Comment (marc-hanheide) [inline app/Models/Student.php]: OK, that's the obvious culprit ;-) Did you make sure none of the functions below are used anywhere?

**Comment by marc-hanheide** [comment]
One problem occured in the deployed test system. Look at https://lcas.lincoln.ac.uk/pgr-test/student/76758442/record/yRxDNZxDwgP/mass-assign and try to assign regulations. It does it but redirects to wrong page. Takes to https://lcas.lincoln.ac.uk/pgr-test/student/76758442/record/509# after assignment

**Comment by marc-hanheide** [comment]
Otherwise it seems to work, but have a play yourself: https://lcas.lincoln.ac.uk/pgr-test/student/76758442/

**Pull request merged**